src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Switch.cs

```diff
@@ -112,29 +112,26 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             var type = ReturnType;
             nodeToCoercedTypeMap = null;
 
-            // Are we on a behavior property?
-            var isBehavior = context.AllowsSideEffects;
-
             // Compute the result type by joining the types of all non-predicate args.
             Contracts.Assert(type == DType.Unknown);
             for (var i = 2; i < count;)
             {
                 var nodeArg = args[i];
                 var typeArg = argTypes[i];
-                if (typeArg.IsError)
-                {
-                    errors.EnsureError(args[i], TexlStrings.ErrTypeError);
-                }
```

Comment:
This check is just moved down to an else if brnach. Since we should be marking fArgsValid to false anytime we encounter error arg, previously it would only mark fArgsValid false when the first arg is error.